extensions/ql-vscode/src/data-extensions-editor/auto-model-codeml-queries.ts

```diff
@@ -228,3 +243,59 @@ export async function runAutoModelQueries({
     candidates,
   };
 }
+
+/**
+ * generateCandidateFilterPack will create a temporary extension pack.
+ * This pack will contain a filter that will restrict the automodel queries
+ * to the specified candidate methods only.
+ * This is done using the `extensible` predicate "automodelCandidateFilter".
+ * @param language
+ * @param candidateMethods
+ * @returns
+ */
+export async function generateCandidateFilterPack(
```

Comment:
Possible performance optimisation: would it make sense to generate the pack in a temp location once up front, at the start of using the data extensions editor, but update the data extensions within the pack with a different set of candidate methods each time you need to run a filter? That might be slightly faster.

Reply:
Any chance that there can be multiple runs happening at the same time? Also, need to make sure that the temp folder is different for each open vscode window.

Reply:
I think we use this approach in a few other places, so I'll defer changes to a followup. I have created an issue to discuss this with the team.